unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -7051,80 +7116,98 @@ public bool objectIsCurrentlyVisible(SimObjPhysics sop, float maxDistance)
+            bool[] standings = null,
+            float? maxDistance = null
+        ) {
+            getInteractablePoses(
+                objectId: objectId,
+                markActionFinished: true,
+                positions: positions, rotations: rotations, horizons: horizons, standings: standings,
+                maxDistance: maxDistance
+            );
         }
-        public void NumberOfPositionsFromWhichItemIsVisible(ServerAction action) {
-            Vector3[] positions = null;
-            if (action.positions != null && action.positions.Count != 0) {
-                positions = action.positions.ToArray();
-            } else {
-                positions = getReachablePositions();
+
+        [ObsoleteAttribute(message: "This action is deprecated. Call GetInteractablePositions instead.", error: false)]
+        public void PositionsFromWhichItemIsInteractable(string objectId, float horizon = 30, Vector3[] positions = null) {
+            // set horizons using the horizon as an increment
+            List<float> horizons = new List<float>();
+            for (float h = -maxUpwardLookAngle; h <= maxDownwardLookAngle; h += horizon) {
+                horizons.Add(h);
             }
+            List<Dictionary<string, object>> interactablePoses = getInteractablePoses(
+                objectId: objectId,
+                markActionFinished: false,
+                positions: positions,
+                horizons: horizons.ToArray()
+            );
 
-            if (!physicsSceneManager.ObjectIdToSimObjPhysics.ContainsKey(action.objectId)) {
-                errorMessage = "Object ID appears to be invalid.";
+            if (interactablePoses == null) {
                 actionFinished(false);
                 return;
             }
 
-            SimObjPhysics theObject = physicsSceneManager.ObjectIdToSimObjPhysics[action.objectId];
+            // for backwards compatibility, PositionsFromWhichItemIsInteractable returns
+            // Dictionary<string, float> instead of List<Dictionary<string, object>>,
+            // where the latter is cleaner in python.
+            Dictionary<string, List<float>> d = new Dictionary<string, List<float>>();
+            string[] keys = {"x", "y", "z", "rotation", "standing", "horizon"};
+            foreach (string key in keys) {
+                d[key] = new List<float>();
+            }
+            foreach(Dictionary<string, object> pose in interactablePoses) {
+                foreach (string key in keys) {
+                    if (key == "standing") {
+                        // standing is converted from true => 1 to false => 0, for backwards compatibility
+                        d[key].Add((bool) pose[key] ? 1 : 0);
+                    } else {
+                        // all other keys have float outputs
+                        d[key].Add((float) pose[key]);
+                    }
+                }
+            }
+            actionFinished(true, d);
```

Comment:
Here, we convert from the cleaner output of `GetInteractablePoses` to the old, backwards compatible, output from `PositionsFromWhichItemIsInteractable`.